commit 9c849ce86e0fa93a218614eac562ace44053d7ce
Merge: e24c567b7ecf 3998f0b8bc49
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Tue Aug 31 09:22:37 2021 -0700

    Merge tag '5.15-rc-smb3-fixes-part1' of git://git.samba.org/sfrench/cifs-2.6
    
    Pull cifs client updates from Steve French:
     "Eleven cifs/smb3 client fixes:
    
       - mostly restructuring to allow disabling less secure algorithms
         (this will allow eventual removing rc4 and md4 from general use in
         the kernel)
    
       - four fixes, including two for stable
    
       - enable r/w support with fscache and cifs.ko
    
      I am working on a larger set of changes (the usual ... multichannel,
      auth and signing improvements), but wanted to get these in earlier to
      reduce chance of merge conflicts later in the merge window"
    
    * tag '5.15-rc-smb3-fixes-part1' of git://git.samba.org/sfrench/cifs-2.6:
      cifs: Do not leak EDEADLK to dgetents64 for STATUS_USER_SESSION_DELETED
      cifs: add cifs_common directory to MAINTAINERS file
      cifs: cifs_md4 convert to SPDX identifier
      cifs: create a MD4 module and switch cifs.ko to use it
      cifs: fork arc4 and create a separate module for it for cifs and other users
      cifs: remove support for NTLM and weaker authentication algorithms
      cifs: enable fscache usage even for files opened as rw
      oid_registry: Add OIDs for missing Spnego auth mechanisms to Macs
      smb3: fix posix extensions mount option
      cifs: fix wrong release in sess_alloc_buffer() failed path
      CIFS: Fix a potencially linear read overflow

diff --cc fs/Kconfig
index 8d6ac56d6fcd,6d719f2c5828..b11bd4b387e1
--- a/fs/Kconfig
+++ b/fs/Kconfig
@@@ -348,8 -358,14 +348,15 @@@ config NFS_V4_2_SSC_HELPE
  
  source "net/sunrpc/Kconfig"
  source "fs/ceph/Kconfig"
+ 
  source "fs/cifs/Kconfig"
 +source "fs/ksmbd/Kconfig"
+ 
+ config CIFS_COMMON
+ 	tristate
+ 	default y if CIFS=y
+ 	default m if CIFS=m
+ 
  source "fs/coda/Kconfig"
  source "fs/afs/Kconfig"
  source "fs/9p/Kconfig"